### PR TITLE
chore/revert btc staking

### PR DESCRIPTION
**Pull request by PaddyMc**
## What is the purpose of the change

Due to a change in design, and after a lot of discussion internally, we've decided this implementation of BTC staking won't bring enough value to the Osmosis ecosystem

We're returning to the design phase of BTC staking, to make sure it's more economically aligned with Osmosis.

The idea behind these PRs means that alloyed BTC would be locked into osmosis and the chain would pay for this security. This isn't the right design for the Osmosis DEX atm, we feel that having a BTC staking that increases TVL and fees is a better trade of than simply paying for security with Omso.

## Testing and Verifying

Do a diff between the superfluid module in v25.x and main and ensure the PRs listed below are reverted.

PRs that are reverted:

- https://github.com/osmosis-labs/osmosis/pull/8320
- https://github.com/osmosis-labs/osmosis/pull/8276
- https://github.com/osmosis-labs/osmosis/pull/8250
- https://github.com/osmosis-labs/osmosis/pull/8169

### CODE REVIEW

**Comment by czarcas7ic** [comment]
Do you think reverting one at a time in reverse order would be worth the overhead? Its kind of hard to reason about it as a whole, but depends on your confidence level on this.

**Comment by PaddyMc** [comment]
I can if that's easier, an alternative is to compare commit diffs to PRs e.g

https://github.com/osmosis-labs/osmosis/pull/8564/commits/f75d11f4da786a7a1a8ca6b6e244b8f6e676c716 => https://github.com/osmosis-labs/osmosis/pull/8320

https://github.com/osmosis-labs/osmosis/pull/8564/commits/2a12368a678fb192a63b1d06d4ea175935a322d6 => https://github.com/osmosis-labs/osmosis/pull/8276

https://github.com/osmosis-labs/osmosis/pull/8564/commits/e1732084e4b8a9eb1ce21522cf0dc0f4ececbd6c => https://github.com/osmosis-labs/osmosis/pull/8250

https://github.com/osmosis-labs/osmosis/pull/8564/commits/7df9cd35be403b21dfe7b5290e8cfe2d99a55b5d => https://github.com/osmosis-labs/osmosis/pull/8169

Also I think doing a diff locally on superfluid on main `c45f95bc517b4f721b7378880803371bc4809803` before the changes and on this branch would be helpful to confirm.

```
diff ./osmosis/x/superfluid ./other/osmosis/x/superfluid
```

```
λ ghost osmosis-labs → diff ./osmosis/x/superfluid ./other/osmosis/x/superfluid    
Common subdirectories: ./osmosis/x/superfluid/client and ./other/osmosis/x/superfluid/client
Common subdirectories: ./osmosis/x/superfluid/keeper and ./other/osmosis/x/superfluid/keeper
diff --color ./osmosis/x/superfluid/module.go ./other/osmosis/x/superfluid/module.go
15c15
< 	abci "github.com/cometbft/cometbft/abci/types"
---
> 	"cosmossdk.io/core/appmodule"
35,36c35,41
< 	_ module.AppModule      = AppModule{}
< 	_ module.AppModuleBasic = AppModuleBasic{}
---
> 	_ module.AppModuleBasic   = AppModuleBasic{}
> 	_ module.HasGenesisBasics = AppModuleBasic{}
> 
> 	_ appmodule.AppModule        = AppModule{}
> 	_ module.HasConsensusVersion = AppModule{}
> 	_ module.HasGenesis          = AppModule{}
> 	_ module.HasServices         = AppModule{}
133a139,144
> // IsAppModule implements the appmodule.AppModule interface.
> func (am AppModule) IsAppModule() {}
> 
> // IsOnePerModuleType is a marker function just indicates that this is a one-per-module type.
> func (am AppModule) IsOnePerModuleType() {}
> 
155c166
< func (am AppModule) InitGenesis(ctx sdk.Context, cdc codec.JSONCodec, gs json.RawMessage) []abci.ValidatorUpdate {
---
> func (am AppModule) InitGenesis(ctx sdk.Context, cdc codec.JSONCodec, gs json.RawMessage) {
161,162d171
< 
< 	return []abci.ValidatorUpdate{}
172c181,182
< func (am AppModule) BeginBlock(ctx sdk.Context, _ abci.RequestBeginBlock) {
---
> func (am AppModule) BeginBlock(context context.Context) error {
> 	ctx := sdk.UnwrapSDKContext(context)
174,179c184
< }
< 
< // EndBlock executes all ABCI EndBlock logic respective to the capability module. It
< // returns no validator updates.
< func (am AppModule) EndBlock(ctx sdk.Context, _ abci.RequestEndBlock) []abci.ValidatorUpdate {
< 	return []abci.ValidatorUpdate{}
---
> 	return nil
205c210
< func (am AppModule) RegisterStoreDecoder(sdr sdk.StoreDecoderRegistry) {
---
> func (am AppModule) RegisterStoreDecoder(sdr simtypes.StoreDecoderRegistry) {
Common subdirectories: ./osmosis/x/superfluid/simulation and ./other/osmosis/x/superfluid/simulation
Common subdirectories: ./osmosis/x/superfluid/types and ./other/osmosis/x/superfluid/types

```

**Review by nicolaslara — APPROVED** [review]

File: x/superfluid/keeper/gov/gov.go [hunk 1]
```diff
@@ -40,7 +37,7 @@ func HandleRemoveSuperfluidAssetsProposal(ctx sdk.Context, k keeper.Keeper, p *t
 			return err
 		}
 		dummyAsset := types.SuperfluidAsset{}
-		if asset.Equal(dummyAsset) {
+		if asset == dummyAsset {
```
Comment (czarcas7ic) [inline x/superfluid/keeper/gov/gov.go:40]: This is confusing to me, shouldn't it have been .Equal( all along?

**Review by czarcas7ic — APPROVED** [review]
Approved but lets confirm on that comment. It wouldn't have been the first time we used == instead of .Equal incorrectly, causing a bug

File: x/superfluid/keeper/gov/gov.go (hunk 1 above)
Comment (PaddyMc) [reply x/superfluid/keeper/gov/gov.go:40]: This was `==` all along, I remember when reviewing it, I'll merge here and create a new PR with this change.

Comment (PaddyMc) [reply x/superfluid/keeper/gov/gov.go:40]: https://github.com/osmosis-labs/osmosis/pull/8580 

:arrow_double_up:

**Pull request merged**